frappe/core/doctype/doctype/doctype.js

```diff
@@ -162,6 +239,43 @@ function render_form_builder_message(frm) {
 	}
 }
 
+function get_conditions_from_field(doctype, fieldname) {
+	const conditions = [
+		["=", __("Equals")],
+		["!=", __("Not Equals")],
+		["like", __("Like")],
+		["not like", __("Not Like")],
+		["in", __("In")],
+		["not in", __("Not In")],
+		["is", __("Is")],
+		[">", ">"],
+		["<", "<"],
+		[">=", ">="],
+		["<=", "<="],
+		["Between", __("Between")],
+		["Timespan", __("Timespan")],
+	];
+	const invalid_conditions_map = {
+		Date: ["like", "not like"],
+		Datetime: ["like", "not like", "in", "not in", "=", "!="],
+		Data: ["Between", "Timespan"],
+		Select: ["like", "not like", "Between", "Timespan"],
+		Link: ["Between", "Timespan", ">", "<", ">=", "<="],
+		Currency: ["Between", "Timespan"],
+		Color: ["Between", "Timespan"],
+		Check: conditions.map((c) => c[0]).filter((c) => c !== "="),
+		Code: ["Between", "Timespan", ">", "<", ">=", "<=", "in", "not in"],
+		"HTML Editor": ["Between", "Timespan", ">", "<", ">=", "<=", "in", "not in"],
+		"Markdown Editor": ["Between", "Timespan", ">", "<", ">=", "<=", "in", "not in"],
+		Password: ["Between", "Timespan", ">", "<", ">=", "<=", "in", "not in"],
+		Rating: ["like", "not like", "Between", "in", "not in", "Timespan"],
+		Float: ["like", "not like", "Between", "in", "not in", "Timespan"],
+	};
+	let current_df = frappe.get_meta(doctype).fields.filter((f) => f.fieldname === fieldname);
+	let invalid_conditions = invalid_conditions_map[current_df[0].fieldtype] || [];
+	return conditions.filter((c) => !invalid_conditions.includes(c[0]));
+}
```

Comment:
Largely a duplicate of [frappe.ui.Filter](https://github.com/frappe/frappe/blob/develop/frappe/public/js/frappe/ui/filters/filter.js).
Do nested_set_conditions – the primary difference between the two – already work?

Reply:
Yes, I am aware.
not as of now, lets see

Reply:
Open to suggestions on how to consolidate them.

Reply:
Well, we might want to add a `nested_set = true` parameter to `frappe.ui.Filter.set_conditions()`.
Were it only a temporary one-off, then this would be more than debatable.
However, this won't be the only situation imaginable where you only want basic operators, no nested set ops.
Guess we'll need a maintainer to weigh in, though.